autopilot/manager.go

```diff
@@ -0,0 +1,277 @@
+package autopilot
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+)
+
+// ManagerCfg houses a set of values and methods that is passed to the Manager
+// for it to properly manage its autopilot agent.
+type ManagerCfg struct {
+	// Self is the public key of the lnd instance. It is used to making
+	// sure the autopilot is not opening channels to itself.
+	Self *btcec.PublicKey
+
+	// PilotCfg is the config of the autopilot agent managed by the
+	// Manager.
+	PilotCfg *Config
+
+	// ChannelState is a function closure that returns the current set of
+	// channels managed by this node.
+	ChannelState func() ([]Channel, error)
+
+	// SubscribeTransactions is used to get a subscription for transactions
+	// relevant to this node's wallet.
+	SubscribeTransactions func() (lnwallet.TransactionSubscription, error)
```

Comment:
Also needs the ability to cancel the subscription? Example scenario being when we turn off autopilot programmatically.

Reply:
It is being done by calling `Cancel` on the returned subscription.